### PR TITLE
report how to create a compiler stacktrace #1280

### PR DESCRIPTION
**Pull request by reactormonk**
How do I get access to the initial command string?

### CODE REVIEW

File: lib/system/excpt.nim [hunk 1]
```diff
@@ -175,6 +175,8 @@ proc auxWriteStackTrace(f: PFrame, s: var string) =
       add(s, tempFrames[j].procname)
     add(s, "\n")
 
+proc stackTraceAvailable*(): bool
```
Comment (Araq) [inline lib/system/excpt.nim]: Remove this. Code size matters.

File: compiler/msgs.nim [hunk 2]
```diff
@@ -715,7 +715,10 @@ type
 proc handleError(msg: TMsgKind, eh: TErrorHandling, s: string) =
   template quit =
     if defined(debug) or gVerbosity >= 3 or msg == errInternal:
-      writeStackTrace()
```
Comment (Araq) [inline compiler/msgs.nim]: use something like:

```nimrod
let st = getStackTrace()
msgWriteln(if st.len != 0: st else: "Use 'koch temp c <file>' to get a stacktrace.")  
```

Comment (reactormonk) [inline compiler/msgs.nim]: Won't work, because `getStackTrace` will return `"No stack traceback available\n"`.

**Comment by reactormonk** [comment]
I don't see any tests explicitly failing because of this change.

File: compiler/msgs.nim (hunk 2 above)
Comment (Araq) [inline compiler/msgs.nim]: So use strutils.countLines(st) > 1: st else: "..." instead. Be more creative please. ;-)

**Pull request merged**